core/shared/src/main/scala/zio/internal/metrics/Gauge.scala

```diff
@@ -0,0 +1,52 @@
+package zio.internal.metrics
```

Comment:
Probably this and other such data types need pulling up into `zio.metrics` because they have safe public APIs that we expect users to use directly.

Reply:
I think we actually may be able to delete this. At this point users interact with metrics through the `ZIOMetric` interface so I'm not sure we need this but perhaps it could be useful to implement variants of `ZIOMetric` for other data types (e.g. `ZStreamMetric`).

Reply:
That would be nice to delete it!